WMFComponents/Sources/WMFComponents/Components/Article Tabs/WMFArticleTabsView.swift

```diff
@@ -20,57 +20,87 @@ public struct WMFArticleTabsView: View {
     /// Flag to allow us to know if we can scroll to the current tab position
     @State private var isReady: Bool = false
     @State private var cellFrames: [String: CGRect] = [:]
+    @State private var hideLoadingWithDelay = false
 
     public init(viewModel: WMFArticleTabsViewModel) {
         self.viewModel = viewModel
     }
 
     public var body: some View {
         GeometryReader { geometry in
-            VStack(spacing: 0) {
-                Group {
-                    if !isReady {
-                        loadingView
-                    } else if viewModel.articleTabs.isEmpty {
-                        emptyStateContainer
-                    } else {
-                        if viewModel.shouldShowTabsV2 {
-                            tabsV2Grid(geometry)
+            ZStack {
```

Comment:
Still a bit nervous with how much we're changing here, I see a loading spinner on control users which doesn't seem quite right to me. Can you create the ZStack and loading overlay only within `tabsV2Grid`? Then hopefully you can do a simple `viewModel.recLoaded` check before adding the overlay instead of all the checks that `bottomDataReady` is doing.